components/ntask/org.wso2.carbon.ntask.core/src/main/java/org/wso2/carbon/ntask/core/impl/AbstractQuartzTaskManager.java

```diff
@@ -218,6 +258,17 @@ protected synchronized void scheduleLocalTask(String taskName,
         }    
     }
 
+    /**
+     * Method to schedule a task with whether or not the task requires recovery set to the specified value.
+     * @param taskName the name of the task to be scheduled
```

Comment:
I suppose with these changes we'll have to do a major version increment given that there are API removals and not completely backward compatible